django-rgd/client/rgd_client/plugin.py

```diff
@@ -118,3 +119,33 @@ def create_file_from_url(
 
         r.raise_for_status()
         return r.json()
+
+    def download_file(self, id: int, download_location: Optional[str] = None) -> Path:
+        """
+        Download a ChecksumFile.
+
+        Args:
+            id: The id of the ChecksumFile
+            download_location: Path to download file to. Defaults to current working directory.
+
+        Returns:
+            The path of the newly downloaded file
+        """
+        r = self.session.get(f'checksum_file/{id}/data')
+        r.raise_for_status()
+
+        if download_location is not None:
+            download_location_path = Path(download_location)
+        else:
+            # Download to current working directory if download_location isn't given
+            download_location_path = Path.cwd()
+
+        # Create the download directory if it doesn't exist
+        download_location_path.mkdir(parents=True, exist_ok=True)
+
+        # Save the file
+        with open(download_location_path / str(id), 'wb') as file:
+            for chunk in r.iter_content(1 << 20):
```

Comment:
Is there a reason for using `1 << 20` (1048576) as the chunk size here? Or is it just convention?

Reply:
Nope, it was just an arbitrary decision to pick 1MB.

Reply:
In that case, maybe just specifying the direct value `1_048_576` and adding a comment to explain that it's 1MB would make things a bit clearer.

Reply:
What is a normal/reasonable value? Is 1MB standard?

Reply:
I believe I've also used 1MB in the past for things like this, imo it's fine.